jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBQueryResultSet.java

```diff
@@ -315,18 +319,16 @@ public boolean getBoolean(int columnIndex) throws SQLException {
 
   @Override
   public boolean getBoolean(String columnName) throws SQLException {
-    String b = getValueByName(columnName);
-    if (b == null) {
+    checkRecord();
+    int index = columnInfoMap.get(columnName) - START_INDEX;
+    Field field = record.getFields().get(index);
+    if (field.getDataType() != null) {
```

Comment:
Is it necessary to check whether the field.gerDataType equals boolean?
If the datatype is not boolean, I guess the field.getBoolV is null too.

Reply:
> Is it necessary to check whether the field.getDataType equals boolean?
> If the datatype is not boolean, I guess the field.getBoolV is null too.

Yes, you are right. It would be better to check whether the field.getDataType is the correct datatype.